src/cascadia/TerminalSettingsModel/FileUtils.cpp

```diff
@@ -39,96 +43,85 @@ namespace winrt::Microsoft::Terminal::Settings::Model
+    // Return Value:
+    // - true if it had the expected permissions. False otherwise.
+    static bool _isOwnedByAdministrators(const HANDLE& handle)
     {
-        // truncate the file because we want to overwrite it
-        SetFilePointer(file.get(), 0, nullptr, FILE_BEGIN);
-        THROW_IF_WIN32_BOOL_FALSE(SetEndOfFile(file.get()));
-
-        const auto fileSize = gsl::narrow<DWORD>(content.size());
-        DWORD bytesWritten = 0;
-        THROW_IF_WIN32_BOOL_FALSE(WriteFile(file.get(), content.data(), fileSize, &bytesWritten, nullptr));
-
-        if (bytesWritten != fileSize)
-        {
-            THROW_WIN32_MSG(ERROR_WRITE_FAULT, "failed to write whole file");
-        }
+        // If the file is owned by the administrators group, trust the
+        // administrators instead of checking the DACL permissions. It's simpler
+        // and more flexible.
+
+        wil::unique_hlocal_security_descriptor sd;
+        PSID psidOwner{ nullptr };
+        // The psidOwner pointer references the security descriptor, so it
+        // doesn't have to be freed separate from sd.
+        const auto status = GetSecurityInfo(handle,
+                                            SE_FILE_OBJECT,
+                                            OWNER_SECURITY_INFORMATION,
+                                            &psidOwner,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            wil::out_param_ptr<PSECURITY_DESCRIPTOR*>(sd));
+        THROW_IF_WIN32_ERROR(status);
+
+        wil::unique_any_psid psidAdmins{ nullptr };
+        THROW_IF_WIN32_BOOL_FALSE(
+            ConvertStringSidToSidW(L"BA", wil::out_param_ptr<PSID*>(psidAdmins)));
+
+        return EqualSid(psidOwner, psidAdmins.get());
     }
-
     // Tries to read a file somewhat atomically without locking it.
     // Strips the UTF8 BOM if it exists.
-    std::string ReadUTF8File(const std::filesystem::path& path)
+    std::string ReadUTF8File(const std::filesystem::path& path, const bool elevatedOnly)
     {
         // From some casual observations we can determine that:
         // * ReadFile() always returns the requested amount of data (unless the file is smaller)
         // * It's unlikely that the file was changed between GetFileSize() and ReadFile()
         // -> Lets add a retry-loop just in case, to not fail if the file size changed while reading.
         for (int i = 0; i < 3; ++i)
         {
-            wil::unique_hfile file{ CreateFileW(path.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, nullptr, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, nullptr) };
+            wil::unique_hfile file{ CreateFileW(path.c_str(),
+                                                GENERIC_READ,
+                                                FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,

```

Comment:
Why is delete OK now?

Reply:
This just broadens the situations in which we can open the file. Widest sharing perms = fewer opportunities for applications to stop us from doing something useful with the user's settings.

Reply:
Most editors overwrite a file when they save, which requires write sharing. Some (vim?) apparently *delete* an existing file first [1]. In Windows 10+, `DeleteFileW()` has been updated to try a POSIX delete. If the filesystem supports it, already open files can continue accessing the original file, even though it's unlinked from the directory. (NTFS renames the deleted file to the reserved directory "\\$Extend\\$Deleted".) If a POSIX delete isn't supported, the file is marked as deleted, but not unlinked, and the name is inaccessible for new opens until all handles for existing opens have been closed. In either case, existing opens have to share delete access. Otherwise `DeleteFileW()` will fail with a sharing violation.

---
[1] IMO, deleting the file is wrong. It discards all of the metadata -- create timestamp, attributes, extended attributes, alternate data streams, and the security descriptor, including the owner, mandatory and discretionary permissions, integrity level, security attributes, and access auditing.